### PR TITLE
Create activities table

### PR DESCRIPTION
**Pull request by crabsinger**
# Create an activities table to build an activities feed

- create database migrations
- setup hasura events
- create event_trigger and test

## Testing:
- create a contribution, add a user, create an epoch, and expect to see the activities table auto populated.
- also added tests for the createActivity file

## Questions:
- we wanted the activities table to hold `actor_profile_id` and `target_profile_id` but that data isn't always passed through, so we need to pass it through or fetch it later.
For users and epochs, we don't have the actor_profile....

## API:

Standard graphql using relations to expose nested activity data types

**Request**
```graphql
query Activities {
  activities {
    epoch {
      id
    }
    action
    user {
      address
      entrance
    }
    contribution {
      description
      id
    }
  }
}
```

**Response**

### CODE REVIEW

File: hasura/migrations/default/1675459146542_create_activities_table/up.sql
Comment (crabsinger) [inline hasura/migrations/default/1675459146542_create_activities_table/up.sql:1]: @levity i dont need you to fully review this code (unless you want to) but since this is a big new DB architecture, I invite your feedback and thoughts before this is merged as its much easier to incorporate changes earlier rather than later.

File: api-test/hasura/event_triggers/createActivity.payloads.ts
Comment (CryptoGraffe) [inline api-test/hasura/event_triggers/createActivity.payloads.ts:1]: is there a better naming convention so that this is obviously for a test?

File: hasura/metadata/databases/default/tables/public_contributions.yaml
Comment (CryptoGraffe) [inline hasura/metadata/databases/default/tables/public_contributions.yaml:93]: might be nice to rename all these to have activity in the front so they are grouped in the UI

File: hasura/migrations/default/1675459146542_create_activities_table/up.sql
Comment (CryptoGraffe) [inline hasura/migrations/default/1675459146542_create_activities_table/up.sql:17]: If we are going to cascade (vs restrict) perhaps we should just delete this activity instead of orphan it? idk

File: api-test/hasura/event_triggers/createActivity.payloads.ts
Comment (CryptoGraffe) [reply api-test/hasura/event_triggers/createActivity.payloads.ts:1]: maybe even a subdirectory

File: hasura/migrations/default/1675459146542_create_activities_table/up.sql
Comment (levity) [reply hasura/migrations/default/1675459146542_create_activities_table/up.sql:1]: this looks good to me -- i can see that avoiding more exotic polymorphism patterns would probably be most compatible with Hasura, and also GraphQL generally -- it seems like [the spec is still in flux](https://github.com/graphql/graphql-spec/issues?q=polymorphism+is%3Aopen)

File: api-lib/gql/mutations.ts
Comment (levity) [inline api-lib/gql/mutations.ts:537]: same comment here as in `api-lib/gql/queries`... actually, in this case, the mutation is so straightforward i wouldn't bother splitting it out into a separate function at all

File: api-lib/gql/queries.ts
Comment (levity) [inline api-lib/gql/queries.ts:83]: please avoid unnecessary newlines where possible

File: hasura/migrations/default/1675459146542_create_activities_table/up.sql
Comment (levity) [inline hasura/migrations/default/1675459146542_create_activities_table/up.sql:11]: this field needs stricter validation -- it should fail to insert if it isn't in a specified set of strings

**Review by levity — CHANGES_REQUESTED** [review]

File: hasura/metadata/databases/default/tables/public_contributions.yaml
Comment (crabsinger) [reply hasura/metadata/databases/default/tables/public_contributions.yaml:93]: ok, how about we name the trigger `activityHandler` since we probably will want to support more than just update in that same file.

and then we can rename these to `activityHandlerContributionInsert`

how's that sound?

File: hasura/migrations/default/1675459146542_create_activities_table/up.sql
Comment (crabsinger) [reply hasura/migrations/default/1675459146542_create_activities_table/up.sql:17]: yeah, that sounds reasonable. I've updated this to cascade deletes.

File: api-lib/gql/queries.ts
Comment (crabsinger) [reply api-lib/gql/queries.ts:66]: done

File: api-test/hasura/event_triggers/createActivity.test.ts
Comment (crabsinger) [reply api-test/hasura/event_triggers/createActivity.test.ts:17]: ok, yeah, this output is plenty nice:

Screenshot 2023-02-15 at 9 31 04 AM

File: api/hasura/event_triggers/eventManager.ts
Comment (crabsinger) [reply api/hasura/event_triggers/eventManager.ts:28]: 1 handler, but three event_triggers that fire that same handler. each trigger is tied to a table action "contribution_insert" or something like that. We're going to eventually have a lot of triggers, but hopefully only one handler. @levity

File: hasura/migrations/default/1675459146542_create_activities_table/up.sql
Comment (crabsinger) [reply hasura/migrations/default/1675459146542_create_activities_table/up.sql:11]: talked with graffe - holding off on this for now.

**Comment by crabsinger** [comment]
Ok, addressed all the feedback. I think this is ready to merge. I think our types are a bit weird right now, as I am seeing some diffs with `yarn generate` around GuildInfoInput added and Discord stuff being removed... so not sure about that...

File: api/hasura/event_triggers/eventManager.ts
Comment (levity) [reply api/hasura/event_triggers/eventManager.ts]: what i meant was, is it possible to define a single handler here named `actvityHandler` and have all three new `event_triggers` YAML sections all refer to that same handler with 
```diff
-  - name: activityHandlerContributionInsert
+  - name: activityHandler
``` 

but i just checked the docs and the event trigger name has to be unique, i guess 🤷

File: api-lib/event_triggers/activity/activityHandler.ts
Comment (levity) [inline api-lib/event_triggers/activity/activityHandler.ts:1]: now that you've created a folder named `activity`, having the files also be prefixed with `activity` seems redundant to me, and none of the other files have `Handler` in their name, so how about renaming to `index.ts` and `mutations.ts`?

Comment (crabsinger) [reply api-lib/event_triggers/activity/activityHandler.ts]: ok, addressed all the feedback.

**Review by levity — DISMISSED** [review]

**Review by CryptoGraffe — APPROVED** [review]

**Review by CryptoGraffe — APPROVED** [review]

**Pull request merged**
